pkg/ingester/ingester_v2.go

```diff
@@ -1501,7 +1503,14 @@ func (i *Ingester) shipBlocks(ctx context.Context) {
 			}
 		}
 
-		// Run the shipper's Sync() to upload unshipped blocks.
+		// Run the shipper's Sync() to upload unshipped blocks. Make sure the TSDB state is active, in order to
+		// avoid any race condition with closing idle TSDBs.
+		if !userDB.casState(active, activeShipping) {
+			level.Warn(util.Logger).Log("msg", "shipper shipped because the TSDB is not active", "user", userID)
```

Comment:
typo: "shipper skipped ..."

Should this be a warning? Looks more like a debug message to me.

Reply:
> typo: "shipper skipped ..."

Ops, fixed!

> Should this be a warning? Looks more like a debug message to me.

Let's find a common ground. I lowered it to Info, but I believe it's worth to notice and most users don't run Cortex with debug log level.

Reply:
My reasoning for "debug" level is that there are only two reasons how this can happen:
- during force-compaction, In this case, after force-compaction, flushing will call shipper anyway
- when tsdb is closed (or being closed), in this case we already know that shipper has nothing to ship

I think that in both cases, fact that shipper has skipped its cycle is not interesting.